### DISCUSSION TITLE
Cannot add custom keys to the localization object without type errors

### DISCUSSION THREAD

**Discussion by rossmeredith** [KevinVandy/mantine-react-table#434] [Q&A]
### mantine-react-table version

2.0.0-beta.7

### react & react-dom versions

18.3.1

### Describe the bug and the steps to reproduce it

It's simple.

If you add a custom key it shows a type error - "Object literal may only specify known properties ..."

How can I overcome this problem?

```

// e.g.

    localization: {
      ...MRT_Localization_EN,
      'exact': 'Exact'
    },

```

### Minimal, Reproducible Example - (Optional, but Recommended)

N/A

### Screenshots or Videos (Optional)

![Screenshot 2024-10-20 103218](https://github.com/user-attachments/assets/0e94f603-17a7-4d83-8ee6-1928053b2ad2)

### Do you intend to try to help solve this bug with your own PR?

No, because I do not know how

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by alessandrojcm** [comment]
What's your use case here? This is meant to replace strings in MRT components, if you need a new string you should probably add it to a custom component.

**Comment by rossmeredith** [comment]
Thanks for responding.

Specifically this is the use case at the moment.  I want to add my own filters.

![image](https://github.com/user-attachments/assets/7ce276c8-b8f5-422b-b1d6-b261538ef5aa)

**Comment by rossmeredith** [comment]
I suppose I could work around this but doesn't it make sense for all translations to be in the same `localization` object?

**Comment by alessandrojcm** [comment]
i18n support is very basic (just a big plain object), this is by design because we don't want to constrain the users into any i18n solution. So if you need anything on top you would have to implement it in userland. FWIW, in my case I use `i18next`, and I have custom components that I add to the table that use the `useTranslation` hook.
